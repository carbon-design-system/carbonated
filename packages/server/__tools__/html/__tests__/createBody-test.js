/**
 * @jest-environment node
 */

'use strict';

describe('createBody', () => {
  let mockManifest;
  let createBody;

  beforeEach(() => {
    mockManifest = {
      'vendor.js': 'vendor.js',
      'main.js': 'main.js',
    };
    createBody = require('../createBody');
  });

  it('should generate the body for the html response', () => {
    expect(createBody(mockManifest)).toMatchSnapshot();
  });

  it('should generate the script tags for the vendor bundle', () => {
    const body = createBody(mockManifest);

    expect(body).toEqual(
      expect.stringContaining(
        `<script crossorigin="anonymous" src="${
          mockManifest['vendor.js']
        }"></script>`
      )
    );
  });

  it('should generate the script tags for the main bundle', () => {
    const body = createBody(mockManifest);

    expect(body).toEqual(
      expect.stringContaining(
        `<script crossorigin="anonymous" src="${
          mockManifest['main.js']
        }"></script>`
      )
    );
  });
});
