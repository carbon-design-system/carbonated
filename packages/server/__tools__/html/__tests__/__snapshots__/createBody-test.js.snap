// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createBody should generate the body for the html response 1`] = `"<body><noscript>You need to enable JavaScript to run this app.</noscript><div id=\\"root\\"></div><script crossorigin=\\"anonymous\\" src=\\"vendor.js\\"></script><script crossorigin=\\"anonymous\\" src=\\"main.js\\"></script></body>"`;
